apps/ledger-live-mobile/src/screens/Portfolio/index.tsx

```diff
@@ -96,89 +75,37 @@ function PortfolioScreen({ navigation }: NavigationProps) {
   const refreshAccountsOrdering = useRefreshAccountsOrdering();
   useFocusEffect(refreshAccountsOrdering);
 
-  const [graphCardEndPosition, setGraphCardEndPosition] = useState(0);
-  const currentPositionY = useSharedValue(0);
-
-  const onPortfolioCardLayout = useCallback((event: LayoutChangeEvent) => {
-    const { y, height } = event.nativeEvent.layout;
-    setGraphCardEndPosition(y + height / 10);
-  }, []);
-
-  const goToAssets = useCallback(() => {
-    navigation.navigate(NavigatorName.Accounts, {
-      screen: ScreenName.Assets,
-    });
-  }, [navigation]);
-
-  const areAccountsEmpty = useMemo(
-    () =>
-      distribution.list &&
-      distribution.list.every(currencyDistribution =>
-        currencyDistribution.accounts.every(isAccountEmpty),
-      ),
-    [distribution],
+  const hasTokenAccounts = useSelector(hasTokenAccountsNotBlacklistedSelector);
+  const hasNonTokenAccounts = useSelector(hasNonTokenAccountsSelector);
+  const hasTokenAccountsWithPositiveBalance = useSelector(
+    hasTokenAccountsNotBlackListedWithPositiveBalanceSelector,
   );
 
-  const [showAssets, assetsToDisplay] = useMemo(
-    () => [
-      distribution.isAvailable && distribution.list.length > 0,
-      distribution.list
-        .filter(asset => {
-          return (
-            asset.currency.type !== "TokenCurrency" ||
-            !blacklistedTokenIds.includes(asset.currency.id)
-          );
-        })
-        .slice(0, maxAssetsToDisplay),
-    ],
-    [distribution, blacklistedTokenIds],
-  );
+  const showAssets =
+    hasNonTokenAccounts || // always show accounts even if they are empty
+    hasTokenAccountsWithPositiveBalance || // always show token accounts if they are not empty
+    (!hideEmptyTokenAccount && hasTokenAccounts); // conditionally show empty token accounts
 
   const data = useMemo(
     () => [
       <Flex px={6} py={4}>
         <FirmwareUpdateBanner />
       </Flex>,
-      <Box mt={3} onLayout={onPortfolioCardLayout}>
-        <GraphCardContainer
-          counterValueCurrency={counterValueCurrency}
-          portfolio={portfolio}
-          areAccountsEmpty={areAccountsEmpty}
-          showGraphCard={showAssets}
-          currentPositionY={currentPositionY}
-          graphCardEndPosition={graphCardEndPosition}
+      <PortfolioGraphCard showAssets={showAssets} />,
+      showAssets ? (
+        <Box background={colors.background.main} px={6} mt={6}>
+          <PortfolioAssets
+            hideEmptyTokenAccount={hideEmptyTokenAccount}
+            openAddModal={openAddModal}
+          />
+        </Box>
+      ) : (
+        <TrackScreen
+          category="Wallet"
+          accountsLength={0}
+          discreet={discreetMode}
         />
```

Comment:
In case `showAssets===true`, the `TrackScreen` that has the `accountsLength` has been moved inside `PortfolioAssets`. This avoids a dependency of this entire screen on useDistribution (which itself depends on `useSelector(accountsSelector)` so it is updated very frequently).